modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/lbs_discrete_ordinates_solver.cc

```diff
@@ -108,10 +108,9 @@ DiscreteOrdinatesSolver::Initialize()
 {
   LBSSolver::Initialize();
 
-  auto src_function = std::make_shared<SourceFunction>(*this);
-
   // Initialize source func
   using namespace std::placeholders;
+  auto src_function = std::make_shared<SourceFunction>(*this);
```

Comment:
another request for some education here. (the *this)

Reply:
`this` is a pointer to the current instance of the object you are in. In this case, the `DiscreteOrdinatesAdjointSolver`. The `*` dereferences it and makes it the actual object. Constructing `SourceFunction` requires a reference to an `LBSSolver` (or descendant).